pubspec.yaml

```diff
@@ -10,6 +10,7 @@ dependencies:
   async: ^2.8.2
   audioplayers: ^1.0.0-rc.4
   badges: ^2.0.2
+  build: ^2.3.0
```

Comment:
@SleepySquash this and `yaml` are only are used in `pubspec_builder.dart` generator. Shouldn't they be in `dev_dependencies`? We don't need them in runtime.

Reply:
@tyranron, we might do this, but analyzer will complain:
```
mayday@MBP messenger-next % flutter analyze
Analyzing messenger-next...                                             

   info • Depend on referenced packages • lib/util/pubspec_builder.dart:19:8 • depend_on_referenced_packages
   info • Depend on referenced packages • lib/util/pubspec_builder.dart:21:8 • depend_on_referenced_packages

2 issues found. (ran in 3.6s)
```

However, `make gen` works just fine, so we might add `// ignore_for_file: depend_on_referenced_packages` at the beginning of the `pubspec_builder.dart`.

Reply:
@SleepySquash maybe it considers `pubspec_builder.dart` as a source file, but it shouldn't. Does it mean it's misplaced?

Reply:
@tyranron it should be placed under `lib/` in order for it to be visible to the `build_runner`:
```
  pubspec_builder:
    import: "package:messenger/util/pubspec_builder.dart"
    builder_factories: ["pubspecBuilder"]
    build_extensions: {"$package$": [".g.dart"]}
    auto_apply: dependents
    build_to: source
```

I might propose an idea to refactor this builder into separate `pub.dev` package, it shouldn't take much time. Not sure if it worth it.

Reply:
@SleepySquash that's strange, so why analyzer complains then? Maybe there is another annotation or a naming convention for the analyzer, so it understands that it's a builder, not a runtime source? If no, let's just place the ignoring thing and that's it.